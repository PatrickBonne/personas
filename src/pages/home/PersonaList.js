import React from 'react'

const PersonaList = ({ children, personaChanged }) => {
  return (
    <ul>
      { React.Children.map(children, (persona) => <li>{React.cloneElement(persona, { personaChanged })}</li>) }
    </ul>
  )
}

export { PersonaList }
