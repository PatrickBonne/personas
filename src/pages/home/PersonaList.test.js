import React from 'react'
import renderer from 'react-test-renderer'
import { PersonaList } from './PersonaList'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <PersonaList items={['P1', 'P2']} personaChanged={() => {}} />
    ).toJSON()
  expect(tree).toMatchSnapshot()
})
