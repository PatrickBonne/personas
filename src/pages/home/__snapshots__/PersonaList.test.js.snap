// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    Names
  </h1>
  <ul>
    <li>
      <button
        onClick={[Function]}
      >
        P1
      </button>
       
    </li>
    <li>
      <button
        onClick={[Function]}
      >
        P2
      </button>
       
    </li>
  </ul>
</div>
`;
